### ISSUE TITLE
Context information is lost in final report when running in mocha parallel mode

### ISSUE THREAD

**Issue by aubertaa** [adamgruber/mochawesome#339]
**Describe the bug**
Using addContext has no effect in final report when running tests in parallel using mocha 'parallel' mode

**Code Reproduce**
run test in parallel mode with code to add context in report file. For instance :

```
const addContext = require("mochawesome/addContext")
...
describe("My testsuite", function () {

            beforeEach(function () {
                addContext(this, "this is a contextual info");
            });

            it("my test", function (done) {
                expect(true).to.be.true;
            });

});
```
**Expected behavior**
Context should be added in testcase when expanding test in HTML report, this works well when we run tests not in parallel mode (--parallel)

**Comment by olexandr13** [comment]
Have the same issue!

**Comment by JakobsenA-ELS** [comment]
Try running Mocha with `--require mochawesome/register`.

**Comment by MihailPertsev** [comment]
I can confirm that using `require("mochawesome/register");` inside .test.js file is solving the problem.
It would be really nice if it worked without that tho...

**Comment by olexandr13** [comment]
> --require mochawesome/register

It helps. Thanks, man!

**Comment by Serg-develop** [comment]
You're amazing, guys)

**Comment by kolbasik** [comment]
You do not need to use `require("mochawesome/register");` in the all tests. There is a simpler way to make it happen:

```
mocha -r mochawesome/register
```

**Comment by joo920131** [comment]
It's not working for me :(

Here's the run with --parallel flag - hooks are not logging the context

![image](https://user-images.githubusercontent.com/42609860/144521763-0a1a44bc-9765-4524-a546-0b42ef86a4ad.png)


And here's the one without - hooks are logging the context this time

![image](https://user-images.githubusercontent.com/42609860/144521817-e45d3800-a425-4dba-a7b4-c97396426d48.png)

Is there something else I need to do particularly for typescript ?

**Comment by kolbasik** [comment]
Hi @joo920131,

I have restored your example in a demo project but I could not reproduce the issue of not attaching context in parallel mode.

[mochawesome-issue-339.zip](https://github.com/adamgruber/mochawesome/files/7694210/mochawesome-issue-339.zip)

Could you please provide the code to reproduce? Thanks.

**Comment by sshaar08** [comment]
the above commandline requirement worked for me thanks! might be worth adding to the docs

**Comment by adamgruber** [comment]
PRs welcome

**Comment by adamgruber** [comment]
Some resolution here: in parallel mode, mocha runs tests in worker processes and serializes their data back to the main process to build the report. `addContext` data (and skipped-test counts) only survive that serialization when `mochawesome/register` is loaded — that requirement is inherent to how parallel mode works, so mochawesome can't make it "just work" without the hook.

What's changed is that the failure is no longer silent: the reporter now **warns** when it detects parallel mode without the register hook, and the README's Parallel Mode section explains why the hook is needed. So the common case in this thread — context missing with no indication why — now points you straight at the fix (`--require mochawesome/register`).

Shipped on `main` and will go out in the next release. Closing as addressed; if you're using the register hook and still see context dropped in parallel mode, please open a new issue with a minimal repro.

---
_Triaged and posted via Claude Code, on behalf of @adamgruber._
